### DISCUSSION TITLE
The logic of work the <script setup> in the layouts

### DISCUSSION THREAD

**Discussion by ruslankonev** [nuxt/nuxt#18272] [Questions]
Anybody know why block <script setup> in Layouts is executed every time when the url is changed?

**Comment by kissu** [comment]
Can you please provide more info or even a repro on this one? 🤗

**Reply by wobsoriano** [reply]
Prolly this https://github.com/nuxt/framework/discussions/1343

**Reply by ruslankonev** [reply]
Yes, this question refers to nuxt/nuxt#18223 and is about why code in <script setup> in layouts is called every time when route changed. According to the docs, <script setup> is called on the server and client to perform the component. This is fine. But why this block of code is called every time in layouts I don't understand
